src/main/java/com/Route.java

```diff
@@ -9,19 +9,21 @@
     private Door endDoor;
     private double lengthInFeet;
     private final List<Node> route;
+    private String errorMsg;
 
-    public Route(List<Node> route, Door start, Door end){
+    public Route(List<Node> route, Door start, Door end, String errorMsg){
```

Comment:
This is fine, but in the future, considering something like keeping this method intact and making a new constructor.

```public Route(List<Node> route, Door start, Door end){ ... }
public Route(List<Node> route, Door start, Door end, String errorMsg)
{
   this.errorMsg = errorMsg;
}
```